src/Spryker/Zed/EventBehavior/Persistence/Propel/Behavior/EventBehavior.php

```diff
@@ -493,6 +504,102 @@ protected function isEventColumnsModified()
         ";
     }
 
+    /**
+     * @return string
+     */
+    protected function addGetOriginalValuesMethod()
+    {
+        $tableName = $this->getTable()->getName();
+        $originalValueColumns = $this->getKeepOriginalValueColumns();
+        $implodedOriginalValueColumns = implode("\n", array_map(function ($columnName) {
+            return sprintf("\t'%s',", $columnName);
+        }, $originalValueColumns));
+
+        return "
+/**
+ * @return array
+ */
+protected function getOriginalValueColumns(): array
+{
+    return [
+    $implodedOriginalValueColumns
+    ];
+}
+
+/**
+ * @return array
+ */
+protected function getOriginalValues(): array
+{
+    if (\$this->isNew()) {
+        return [];
+    }
+
+    \$originalValues = [];
+    foreach (\$this->_modifiedColumns as \$modifiedColumn) {            
+        if (!in_array(\$modifiedColumn, \$this->getOriginalValueColumns())) {
+            continue;
+        }
+
+        \$before = \$this->_initialValues[\$modifiedColumn];
+        \$field = str_replace('$tableName.', '', \$modifiedColumn);
+        \$after = \$this->\$field;
+        
+        if (\$before !== \$after) {
+            \$originalValues[\$modifiedColumn] = \$before;
+        }
+    }
+
+    return \$originalValues;
+}        
+        ";
+    }
+
+    /**
+     * @return array
+     */
+    protected function getKeepOriginalValueColumns(): array
```

Comment:
please use the better name for this method `getKeepOriginalValueColumnNames` or `getAffectedColumnNames`

Reply:
Done.